packages/react-server-cli/src/compileClient.js

```diff
@@ -0,0 +1,216 @@
+		output: {
+			path: outputDir,
+			publicPath: outputUrl,
+			filename: "[name].bundle.js",
+			chunkFilename: "[id].bundle.js",
+		},
+		module: {
+			loaders: [
+				{
+					test: /\.jsx?$/,
+					loader: "babel",
+					exclude: /node_modules/,
+				},
+				{
+					test: /.css$/,
+					loader: extractTextLoader,
+					exclude: /node_modules/,
+				},
+			],
+		},
+		plugins: [
+			new ExtractTextPlugin("[name].css"),
+		],
+	};
+
+	if (minify) {
+		webpackConfig.plugins = [
+			...webpackConfig.plugins,
+			new webpack.DefinePlugin({
+				'process.env': {NODE_ENV: '"production"'},
+			}),
+			// TODO: should this be done as babel plugin?
+			new webpack.optimize.UglifyJsPlugin(),
+		];
+	} else {
+		webpackConfig.devtool = "#cheap-module-eval-source-map";
+	}
+
+	if (hot) {
+		webpackConfig.module.loaders.unshift({
+			test: /\.jsx?$/,
+			loader: "react-hot",
+			exclude: /node_modules/,
+		});
+		webpackConfig.plugins = [
+			...webpackConfig.plugins,
+			new webpack.HotModuleReplacementPlugin(),
+			new webpack.NoErrorsPlugin(),
+		];
+	}
+
+	return webpackConfig;
+};
+
+// writes out a routes file that can be used at runtime.
+const writeWebpackCompatibleRoutesFile = (routes, routesDir, workingDirAbsolute, staticUrl, isClient) => {
+	let routesOutput = [];
+
+	const existingMiddleware = routes.middleware ? routes.middleware.map((middlewareRelativePath) => {
+		return `unwrapEs6Module(require("${path.relative(workingDirAbsolute, path.resolve(routesDir, middlewareRelativePath))}"))`
+	}) : [];
+	routesOutput.push(`
+function unwrapEs6Module(module) { return module.__esModule ? module.default : module }
+var coreJsMiddleware = unwrapEs6Module(require('react-server-cli/target/coreJsMiddleware'));
+var coreCssMiddleware = unwrapEs6Module(require('react-server-cli/target/coreCssMiddleware'));
+module.exports = {
+	middleware:[
+		coreJsMiddleware(${JSON.stringify(staticUrl)}),
+		coreCssMiddleware(${JSON.stringify(staticUrl)}),
+		${existingMiddleware.join(",")}
+	],
+	routes:{`);
+
+	for (let routeName of Object.keys(routes.routes)) {
+		let route = routes.routes[routeName];
+
+		routesOutput.push(`
+		${routeName}: {`);
+		for (let name of ["path", "method"]) {
+			routesOutput.push(`
+			${name}: "${route[name]}",`);
+		}
+
+		let formats = normalizeRoutesPage(route.page);
+		routesOutput.push(`
+			page: {`);
+		for (let format of Object.keys(formats)) {
+			const formatModule = formats[format];
+			var relativePathToPage = path.relative(workingDirAbsolute, path.resolve(routesDir, formatModule));
+			routesOutput.push(`
+				${format}: function() {
+					return {
+						done: function(cb) {`);
+			if (isClient) {
+				routesOutput.push(`
+							require.ensure("${relativePathToPage}", function() {
+								cb(unwrapEs6Module(require("${relativePathToPage}")));
+							});`);
+			} else {
+				routesOutput.push(`
+							cb(unwrapEs6Module(require("${relativePathToPage}")));`);
+			}
+			routesOutput.push(`
+						}
+					};
+				},`);
+		}
```

Comment:
Wow... That's... wow.  Braces.  I bet the output of this is nicely indented, huh?